torch/_higher_order_ops/cond.py

```diff
@@ -143,13 +147,186 @@ def _validate_input(pred, true_fn, false_fn, operands):
                     pred, true_fn, false_fn, operands
                 )
 
-
 """
 We're going to define a `cond_op` operation.
 In order to do this, we need implementations for each of the dispatch keys.
 """
 cond_op = HigherOrderOperator("cond")
 
+dummy_aot_config = AOTConfig(
+    fw_compiler=None,  # type: ignore[arg-type]
+    bw_compiler=None,  # type: ignore[arg-type]
+    partition_fn=None,  # type: ignore[arg-type]
+    decompositions={},
+    num_params_buffers=0,
+    aot_id=0,
+    keep_inference_input_mutations=False,
+)
+
+
+def create_fw_bw_graph(true_fn, false_fn, operands):
+
+    # Note:[HOP create fw_bw graph] We create "clean" environments for make_fx by suspending all dispatch keys
+    # between Autograd and Python key. Currently, we only suspend functionalization but more can be
+    # added when required. Will encounter two problems if we don't suspend functionalization:
```

Comment:
This redundancy with map is not ideal. We should probably think about ways of unifying the implementation into a helper function and use the same helper function for control flow ops.

Reply:
This test still fails, as the graph module does not contain the backward graph. I have realized that the root cause of this is that in the [forward function of CondAutogradOp](https://github.com/pytorch/pytorch/pull/126911/files#diff-4093d24068320f82997aecfaf4260aa495e3b4322e178cb5625f9555e3f8b0f5R359-R363), the `cond_op` is torch.compiled.

Reply:
I consolidated some of the commonly used methods and put them into the [utils.py](https://github.com/pytorch/pytorch/pull/126911/files#diff-94838db6991b14cfeec36d4eb0e0ff6fadc08624a7c3ac967820f7f7ae267e70R205-R300)